libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo3/ast/internal/ExecutableValidationScope.kt

```diff
@@ -282,13 +274,10 @@ internal class ExecutableValidationScope(
       return
     }
 
-    fragmentDefinition.selections.validate(fragmentTypeDefinition, this@validate, path)
```

Comment:
Most of the validations do not recurse in fragment spreads now. Exceptions are 

* `@defer` validations
* fieldsInSetCanMerge

They are handled separately